sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -142,6 +171,17 @@
   }
 
   /**
+   * Like {@link #readGenericRecords(Schema)}, but reads each filepattern in the input {@link
+   * PCollection}.
+   */
+  public static ReadAll<GenericRecord> readAllGenericRecords(Schema schema) {
```

Comment:
Would be nice but I don't know how to do it because AvroCoder doesn't do it. It's definitely outside the scope of the current PR, though it is useful future work if we figure out how to do it.